feat: promisified fs.stat and fs.readdir and handled the returned promise

hrithik-b01 opened the pull request:
I have promisified the fs.stat and fs.readdir API's. Using promisified fs APIs, I have handled the returning promise from these functions, and added then() and catch() blocks to process the returned value and display any error respectively. I have tested the function in the same file. Please review my code, and provide feedbacks on how I can further improve my code, or if any errors, please let me know.

shripada commented on src/promises.js:
```diff
@@ -0,0 +1,89 @@
+      if (stats.isFile()) {
+        totalSize += stats.size;
+      } else if (stats.isDirectory()) {
+        fs.readdirSync(folderPath).forEach((file) => {
```
Can you implement the same using the promisified versions of these `fs` apis?

hrithik-b01 commented:
> I have promisified the fs.stat and fs.readdir API's. Using promisified fs APIs, I have handled the returning promise from these functions, and added then() and catch() blocks to process the returned value and display any error respectively. I have tested the function in the same file. Please review my code, and provide feedbacks on how I can further improve my code, or if any errors, please let me know.

I have used **reduce()** higher order function to make a custom implementation of the **Promise.All()** api. This function accepts an array of promises, and returns a single promise, after consuming all the promises inside the array. I have also added tests. Please review my code and provide feedbacks on how I can improve the code.
